Scuti/Scripts/UI/Store/Offer Details/UIPanningAndPinchImage.cs

```diff
@@ -0,0 +1,334 @@
+        {
+            Debug.Log("Begin Drag");
+            lastMousePosition = eventData.position;
+            averagePointBetweenTouch = eventData.position;
+        }
+
+        /// <summary>
+        /// This method will be called during the mouse/touch drag
+        /// </summary>
+        /// <param name="eventData">mouse pointer event data</param>
+        public void OnDrag(PointerEventData eventData)
+        {
+            isDragDetected = true;
+            scrollRect.enabled = false;
+
+            Vector2 currentMousePosition = new Vector2(0, 0);
+
+            // Get de position touch
+            if (counterTouch == 1)
+            {
+                currentMousePosition = eventData.position;
+            }
+
+            if (counterTouch >= 2)
+            {
+                currentMousePosition = averagePointBetweenTouch;
+            }
+
+            Vector2 diff = currentMousePosition - lastMousePosition;
+            RectTransform rect = GetComponent<RectTransform>();
+
+            Vector3 newPosition = rect.position + new Vector3(diff.x, diff.y, 0/*transform.position.z*/);
+            Vector3 oldPos = rect.position;
+            rect.position = newPosition;
+
+            lastMousePosition = currentMousePosition;
+
+        }
+
+
+        /// <summary>
+        /// This method will be called at the end of mouse drag
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnEndDrag(PointerEventData eventData)
+        {
+            if (counterTouch == 0)
+            {
+                isDragDetected = false;
+            }
+            scrollRect.enabled = true;           
+
+            Debug.Log("End Drag");
+
+        }
+
+        // ----------------------------------------------------------------------------------------------
+
+        private void MouseDown(PointerEventData dat)
+        {
+            if (dat == null)
+                return;
+
+            Vector2 localCursor;
+            //var rect1 = scrollRect.GetComponent<RectTransform>();
+            var pos1 = dat.position;
+            if (!RectTransformUtility.ScreenPointToLocalPointInRectangle(rectScroll, pos1,
+                null, out localCursor))
+                return;
+
+            int xpos = (int)(localCursor.x);
+            int ypos = (int)(localCursor.y);
+
+            if (xpos < 0) xpos = xpos + (int)rectScroll.rect.width / 2;
+            else xpos += (int)rectScroll.rect.width / 2;
+
+            if (ypos > 0) ypos = ypos + (int)rectScroll.rect.height / 2;
+            else ypos += (int)rectScroll.rect.height / 2;
+
+           //  Debug.Log("Correct Cursor Pos: " + xpos + " " + ypos);
+
+            // Calculate point in scale
+            float widthScroll = rectScroll.rect.width;
+            float heightScroll = rectScroll.rect.height;
+
+            // Debug.Log("Scroll: Width - Height: " + widthScroll + " " + heightScroll);
+
+            var rect2 = GetComponent<RectTransform>();
```

Comment:
Don't call GetComponent... use a reference variable and assign in editor.

Reply:
Done